lib/widgets/message_list.dart

```diff
@@ -161,7 +189,26 @@ class _MessageListState extends State<MessageList> {
           child: Center(
             child: ConstrainedBox(
               constraints: const BoxConstraints(maxWidth: 760),
-              child: _buildListView(context))))));
+              child: NotificationListener<ScrollEndNotification>(
```

Comment:
Huh indeed. Sounds like a bit of a pitfall there in the Flutter API — easy to miss that the second listener is necessary — plus it sounds like using the ScrollMetricsNotification for build may suffer that same one-frame lag, though that's not entirely clear. Better a one-frame lag than a forever lag, though, and I don't see a better way in the existing API, so that combination sounds good.